PWGCF/MultiparticleCorrelations/Core/MuPa-MemberFunctions.h

```diff
@@ -2015,4 +2186,417 @@ Double_t Weight(const Double_t& value,
+      if (!(wPhi > 0.)) {
+        LOGF(error, "\033[1;33m%s wPhi is not positive, skipping this particle for the time being...\033[0m", __PRETTY_FUNCTION__);
+        LOGF(error, "dPhi = %f\nwPhi = %f", dPhi, wPhi);
+        continue;
+      }
+    } // if(pw_a.fUseWeights[wPHI])
+    if (pw_a.fUseWeights[wPT]) {
+      wPt = Weight(dPt, "pt"); // corresponding pt weight
+      if (!(wPt > 0.)) {
+        LOGF(error, "\033[1;33m%s wPt is not positive, skipping this particle for the time being...\033[0m", __PRETTY_FUNCTION__);
+        LOGF(error, "dPt = %f\nwPt = %f", dPt, wPt);
+        continue;
+      }
+    } // if(pw_a.fUseWeights[wPT])
+    if (pw_a.fUseWeights[wETA]) {
+      wEta = Weight(dEta, "eta"); // corresponding eta weight
+      if (!(wEta > 0.)) {
+        LOGF(error, "\033[1;33m%s wEta is not positive, skipping this particle for the time being...\033[0m", __PRETTY_FUNCTION__);
+        LOGF(error, "dEta = %f\nwEta = %f", dEta, wEta);
+        continue;
+      }
+    } // if(pw_a.fUseWeights[wETA])
+
+    for (Int_t h = 0; h < gMaxHarmonic * gMaxCorrelator + 1; h++) {
+      for (Int_t wp = 0; wp < gMaxCorrelator + 1; wp++) { // weight power
+        if (pw_a.fUseWeights[wPHI] || pw_a.fUseWeights[wPT] || pw_a.fUseWeights[wETA]) {
+          wToPowerP = pow(wPhi * wPt * wEta, wp);
+        }
+        qv_a.fQvector[h][wp] += TComplex(wToPowerP * TMath::Cos(h * dPhi), wToPowerP * TMath::Sin(h * dPhi));
+      } // for(Int_t wp=0;wp<gMaxCorrelator+1;wp++)
+    }   // for(Int_t h=0;h<gMaxHarmonic*gMaxCorrelator+1;h++)
+
+    // *) Nested loops containers:
+    if (fCalculateNestedLoops || fCalculateCustomNestedLoop) {
+      if (nl_a.ftaNestedLoops[0]) {
+        nl_a.ftaNestedLoops[0]->AddAt(dPhi, fSelectedTracks);
+      } // remember that the 2nd argument here must start from 0
+      if (nl_a.ftaNestedLoops[1]) {
+        nl_a.ftaNestedLoops[1]->AddAt(wPhi * wPt * wEta, fSelectedTracks);
+      } // remember that the 2nd argument here must start from 0
+    }   // if(fCalculateNestedLoops||fCalculateCustomNestedLoop)
+
+    // *) Counter of selected tracks in the current event:
+    fSelectedTracks++;
+    if (fSelectedTracks >= cSelectedTracks_max) {
+      break;
+    }
+
+  } // for (auto& track : tracks)
+
+} // void MainLoopOverParticlesRecSim(TracksRecSim const& tracks)
+
+//============================================================
+
+void MainLoopOverParticlesSim(TracksSim const& tracks)
+{
+  // This is the main loop over particles, in which Q-vectors and particle histograms are filled, particle cuts applied, etc.
+  // Only 'sim' info is processed in this loop, set gProcessSim = true via configurable "cfWhatToProcess"
+
+  if (fVerbose) {
+    LOGF(info, "\033[1;32m%s\033[0m", __PRETTY_FUNCTION__);
+  }
+
+  // TBI 20231021 tbc...
+
+} // void MainLoopOverParticlesSim(TracksSim const& tracks)
+
+//============================================================
```

Comment:
Please, don't replicate this code.
I will allow you to progress but I don't want to see it in the next iteration. Grrrrrh!